e-kai00/finFlare
# PULL REQUEST: Start functionality for fetching data and more.

### Description:
This pull request addresses mostly my start functionality for finance data fetching.

### Changes Made:
- Updated authentication middleware
- Added additional validation checks
- Added favicon
- Added fontawesome user icon in navbar
- Changed structure of allauth folders
- Start to create the market template with simple basic stock function to fetch finance API

### Testing:
The fetching of finance api works. But I only have 100 request left. We need to keep track of that, and implement a new API key when this one is over.

1. Run the development server: `python manage.py runserver`.
2. Navigate to the finance data page and ensure that stock information is displayed correctly.


### Related Issues:

### Checklist:

### Additional Notes:
None
